TicTacToe/PlayField.cpp

```diff
@@ -0,0 +1,105 @@
+#include <vector>
+#include <cassert>
+#include "PlayField.h"
+
+using namespace std;
+
+bool PlayField::operator==(const PlayField& second) {
+	for (int x = 0; x < DIM; x++)
+		for (int y = 0; y < DIM; y++)
+			if (matrix[x][y] != second[CellIdx::CreateIndex(x, y)])
+				return false;
+	return true;
+}
+
+PlayField PlayField::operator+(CellIdx index) const {
+	assert(matrix[index.X()][index.Y()] == csEmpty);
+	PlayField newField = PlayField(*this);
+	newField.matrix[index.X()][index.Y()] = GetNextMove();
+	return newField;
+}
+
+PlayField::CellState PlayField::operator[](CellIdx index) const{
+	return matrix[index.X()][index.Y()];
+}
+
+std::vector<PlayField::CellIdx> PlayField::getEmptyCells() const{
+	vector<CellIdx> emptyCells;
+	for (int x = 0; x < DIM; x++)
+		for (int y = 0; y < DIM; y++)
+			if (matrix[x][y] == csEmpty)
+				emptyCells.push_back(PlayField::CellIdx::CreateIndex(x, y));
+	return emptyCells;
+}
+
+PlayField::FieldStatus PlayField::checkFieldStatus() const{
+	int crossesCount = 0;
+	int noughtsCount = 0;
+	Counts(crossesCount, noughtsCount);
+	bool isCrossesWin = HasWinSequence(csCross);
+	bool isNoughtsWin = HasWinSequence(csNought);
+	bool isNormal = abs(crossesCount - noughtsCount) < 2;
+	bool isMovesEnd = getEmptyCells().empty();
+	FieldStatus isWinOrDraw = isCrossesWin ? fsCrossesWin : isNoughtsWin ? fsNoughtsWin : fsDraw;
+	return (isCrossesWin || isNoughtsWin || (isNormal && isMovesEnd)) ? isWinOrDraw : isNormal ? fsNormal : fsInvalid;
+}
+
+PlayField PlayField::makeMove(CellIdx index) const{
+	assert(matrix[index.X()][index.Y()] == csEmpty);
+	return PlayField(*this) + index;
```

Comment:
```suggestion
	return *this + index;
```
Зачем сразу копию делать? operator+ сделает вам копию сам.